support-frontend/app/admin/settings/Switches.scala

```diff
@@ -1,21 +1,121 @@
 package admin.settings
 
 import com.gu.support.encoding.Codec
-import com.gu.support.encoding.Codec.deriveCodec
+import io.circe.generic.extras.Configuration
+import io.circe.generic.extras.auto._
+import io.circe.{Decoder, Encoder, Json}
+import io.circe.optics.JsonPath._
+
+sealed trait SwitchState {
+  def isOn: Boolean
+}
+case object On extends SwitchState {
+  def isOn: Boolean = true
+}
+case object Off extends SwitchState {
+  def isOn: Boolean = false
+}
+
+object SwitchState {
+  import io.circe.generic.extras.semiauto.{ deriveEnumerationDecoder, deriveEnumerationEncoder }
+  implicit val stateDecoder: Decoder[SwitchState] = deriveEnumerationDecoder[SwitchState]
+  implicit val stateEncoder: Encoder[SwitchState] = deriveEnumerationEncoder[SwitchState]
+}
+
+case class FeatureSwitches(
+  enableQuantumMetric: SwitchState = Off,
+  usStripeAccountForSingle: SwitchState = On
+)
+
+case class CampaignSwitches(
+  enableContributionsCampaign: SwitchState = Off,
+  forceContributionsCampaign: SwitchState = Off
+)
+
+case class SubscriptionsSwitches(
+  enableDigitalSubGifting: SwitchState = On,
+  useDotcomContactPage: SwitchState = On,
+  checkoutPostcodeLookup: SwitchState = On
+)
+
+case class RecaptchaSwitches(
+  enableRecaptchaBackend: SwitchState = On,
+  enableRecaptchaFrontend: SwitchState = On
+)
+
+case class OneOffPaymentMethodSwitches(
+  stripe: SwitchState = On,
+  stripeApplePay: SwitchState = On,
+  stripePaymentRequestButton: SwitchState = On,
+  payPal: SwitchState = On,
+  amazonPay: SwitchState = On
+)
+
+case class RecurringPaymentMethodSwitches(
+  stripe: SwitchState = On,
+  stripeApplePay: SwitchState = On,
+  stripePaymentRequestButton: SwitchState = On,
+  payPal: SwitchState = On,
+  directDebit: SwitchState = On,
+  existingCard: SwitchState = On,
+  existingDirectDebit: SwitchState = On,
+  amazonPay: SwitchState = Off,
+  sepa: SwitchState = Off
+)
 
 case class Switches(
-  oneOffPaymentMethods: PaymentMethodsSwitch,
-  recurringPaymentMethods: PaymentMethodsSwitch,
-  enableDigitalSubGifting: SwitchState,
-  useDotcomContactPage: Option[SwitchState],
-  enableRecaptchaBackend: SwitchState,
-  enableRecaptchaFrontend: SwitchState,
-  experiments: Map[String, ExperimentSwitch],
```

Comment:
we no longer have an `experiments` Map. All switches must be properly defined in one of the switch groups in this file